### ISSUE TITLE
The meta system

### ISSUE THREAD

**Issue by alfiejfs** [CyberDiscovery/meta#211]
Sigh, here we are again with yet another issue on the meta system. I thought of this idea earlier, and I thought I would share it.

I would like to preface this by saying that I personally don't like the meta system, but I don't particularly dislike it either - I'm kind of neutral on the topic. I agree that the suggestions system did need more organisation, but I don't really agree that making a meta repository was the way to do it. The meta repository is very organised, however it requires an extensive amount of preparation for what it does - you'll spend more time setting things up than suggesting and implementing things. So that leaves me here (attempting) to provide a solution. From what I've seen, people want the suggestions system to remain in Discord, so here is what I thought of.

All the details can be fine tuned, but you'll see the general idea.

**The system**

You suggest something as normal, using a designated command. However, as opposed to using the other suggestions bot that is now used for polls, you would use a command that has its functionality built-in to the Cyber Discovery bot. It would take your suggestion, and place it in a channel with reactions, like it previously worked. If the suggestion reached a certain amount of reactions, a channel would be made (the description being the suggestion) and everyone could see and talk in the channel. The embed would be re-sent at the top of the channel and pinned as well. From there, people can discuss and fine tune the idea. The creator of the suggestion has access to pin things in the channel. After a certain amount of discussion and consideration, when a conclusion is reached, the mods can archive the channel using another command. It will be sent to an archive (or deleted). 

TL;DR:
 - Suggest something in Discord
 - If people like it, they react and if it gets enough reactions a channel is made
 - People discuss in the channel
 - Mods make a decision and archive channel

I believe this system is good as it allows for discussion, but also keeps the discussion in Discord.

**Possible flaws**

- Abuse would be a big issue. To circumvent this, you could make it Trusted only, and any abuse could warrant the removal of your Trusted role
- Too many channels could also be an issue, as mods could find themselves scrolling endlessly. Well, the solution may not be clear if you want to archive everything, but archiving everything isn't always necessary - you could delete ideas that are denied and silly, or ideas that are implemented with clear reasons. You would only need to keep the channels of ideas that need explanations as to why they were implemented.

**Fine tuning**

You could fine tune this in several ways, some of which I touched on above.

- Who can use the bot
- How many reactions it takes to pass on to discussion
- Should suggestions be deleted after X amount of negative reactions
- Should their be a cooldown on making suggestions

I hope you like the suggestion.

**Comment by lightspeedana** [comment]
Whilst the suggestion sounds quite cool and effective, the implementation sounds very difficult - someone would have to code this part of the bot from scratch and that would certainly take far more time than it took to set up the meta repository. The idea is a cool suggestion, but I don't think it's feasible for implementation, certainly not without a lot of comdev work, also right mid-exam season.

**Comment by alfiejfs** [comment]
@lightspeedana it's doable over time, and doesn't have to be implemented right away. I don't think developer time should be that much of an issue, considering there are many who are willing (or at least I am) to implement this.

**Comment by thebeanogamer** [comment]
So I should preface this by saying that the meta repo in its current form is not the intended final incarnation. The overall plan is to allow for items to go into `#suggestions` style voting once discussion has completed on GitHub, we simply haven’t had the time to implement that yet. Additionally, issue templates are currently being implemented to help streamline the submission process.

My main concern with a system as you describe it would be both the sheer level of channels it would produce. You touched on this yourself, but this would produce an unholy level of channels, and whilst the Trusted role does represent a reasonable level of trust, it is hardly definitive, and this would in effect grant the ability for anyone who had it to create an unlimited number of channels on the server. Doing this through GitHub allows us to have an unlimited number of structured discussions which do not interfere with the server itself.

Additionally, this would be an incredibly large amount of work for us, requiring us to both build a far more complex system than we have at present and grant many more permissions to the bot compared to what it currently has.

We appreciate that there is a wide variety of opinions over this initiative, and we are making changes to reflect some of these wishes. Whilst I don’t personally think this is the solution, it is a sign that there are those with suggestions on how we can help improve this, which can only be a good thing. I’d really like to thank you for your submission as you’ve clearly put a lot of work into it, but in my opinion, it’s not currently within the scope of what is feasible. I am open to hearing both arguments in favour of this proposal or alternatives within a similar vein, and as such will leave the issue open for now.

**Comment by alfiejfs** [comment]
@thebeanogamer yes, that is the only feasible issue I see. You could mitigate this by adding a cooldown, and deleting as opposing to archiving, but the issue would still remain there.

As for your thoughts on the trusted role, and it not representing someone being trusted -- why not make it so? It could be awarded similar to how you would award mentor / moderator: to members who you think deserve it, and won't abuse it. As such, the problem is somewhat mitigated.

Anyhow, thank you for your opinions, and I do agree with what you were saying. Channels is an issue, but with countermeasures in place it can be controlled and condensed.

**Comment by NicholasG04** [comment]
Yeah, I think the idea of holding a new channel for each issue is actually completely ridiculous, even though I like the sound of your ideas.

**Comment by teamshortcut** [comment]
Agreed with @thebeanogamer and @lightspeedana above. I would particularly worry about new members joining, for whom meta discussion will be largely irrelevant; the proposed system, besides being cluttered, would be confusing as all hell for them. I will re-emphasise that feasible suggestions from this repo will go into #meta for voting, so the process is far from confined solely to GitHub.

**Comment by alfiejfs** [comment]
@teamshortcut the issue with the current is not so much that it isn't partially on Discord - no. It's the fact that is partially on Github. 

"I will re-emphasise that feasible suggestions from this repo will go into #meta for voting, so the process is far from confined solely to GitHub."

That means you have to make a GitHub account to make suggestions, which is the issue and problem here. This was an attempt at a solution (which clearly has a huge flaw, which I see), but I don't think lots of channels is worse than a GitHub suggestion system for newer people (which you said your concern would revolve around). 

For a server aimed at 14-18, a development platform that is quite complex to someone with no prior knowledge isn't the best solution at all.

**Comment by teamshortcut** [comment]
The difference is that a large number of channels impacts all members, irrespective of their interest or involvement in the suggestions system. A GitHub account is free and easy to make, and while I accept that it's more effort than a command in Discord, it's really not a huge barrier for participation, particularly given the 14-18 year olds you're referring to are in a server based around cyber security.

**Comment by NicholasG04** [comment]
@AvrooVulcan It should however be noted that due to the nature of the server many members are already utilising Git and the GitHub platform specifically.
